domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmAddHandler.java

```diff
@@ -779,7 +778,7 @@ private void addSecretService(OperationContext context, ModelNode secret, String
         ServiceName secretServiceName = SecretIdentityService.ServiceUtil.createServiceName(realmName);
 
         ModelNode resolvedValueNode = SecretServerIdentityResourceDefinition.VALUE.resolveModelAttribute(context, secret);
-        boolean base64 = secret.get(SecretServerIdentityResourceDefinition.VALUE.getName()).getType() != ModelType.EXPRESSION;
+        boolean base64 = true;
```

Comment:
This seems like a change that breaks compatibility.  AIUI the issue is that the attribute description, and perhaps some other docs, state that the value of the attribute is Base64 encoded, but the logic assumes that the value from a resolved expression would be plain text. Which is a reasonable expectation, as encoding a password in Base64 brings no real security value, whereas putting it in plain text in a vault and resolving it from a vault would be standard vault usage.

Regardless of what the attribute description and the docs say, this behavior has been in the code since at least 2013. I don't think we can just change it to require a Base64 encoded value. No one using an expression would have Base64 encoded the resolved value as it wouldn't have worked.

*Perhaps* we could make base64 as Boolean, not a boolean and pass 'null' if it was an expression. Then SecretIdentityService *tries* to Base64 decode, and just uses the plain text if that fails.  But even then you could have a corner case breaking change where a existing plain text value just happened to be valid Base64 text.

Reply:
If the value of expression is not base64 then anyways it will throw error like below at the starting of JBoss and deployment will be failed . 

----------------------------------------
2019-06-18 16:50:54,966 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-6) MSC000001: Failed to start service org.wildfly.core.management.security.realm.ejb-security-realm.secret: org.jboss.msc.service.StartException in service org.wildfly.core.management.security.realm.ejb-security-realm.secret: Failed to start service 


2019-06-18 16:50:57,296 ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0013: Operation ("deploy") failed - address: ([("deployment" => "TestEJb_RemoteOutbound.war")]) - failure description: {
.
.
2019-06-18 16:50:57,299 ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0013: Operation ("add") failed - address: ([
    ("core-service" => "management"),
    ("security-realm" => "ejb-security-realm")
----------------------------------------


However , I updated the pull request

Reply:
Is there no more error logged.

If more if available it is often helpful to just paste the complete error, often the exception reported contains a simple error message however the detail is really within the nested causes.

Reply:
@iingawal I don't understand how org.jboss.as.domain.management.security.SecretIdentityService.start(SecretIdentityService.java:77 is reached if the of the SecretServerIdentityResourceDefinition.VALUE attribute was an expression.  With the change in this PR it will fail, but I don't see how it would have before.

Reply:
Perhaps my last comment is out of date though as I see now you've added code to check for Base64.